core/shared/utils/uncommon/bh_getopt.c

```diff
@@ -0,0 +1,51 @@
+/* *****************************************************************
+*
+* Copyright 2016 Microsoft
+*
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+******************************************************************/
+#ifdef WIN32
+#include "bh_getopt.h"
+#include <windows.h>
+char* optarg = NULL;
+int optind = 1;
```

Comment:
The two variables arg global, could we move them into getopt as local variables?

Reply:
No, we couldn't. At least optind cannot be changed to local variable, because some place like samples/basic/src/main.c use the optind out of the getopt function.
And also from the getopt(3) man page:

> SYNOPSIS
>
>      #include <unistd.h>
> 
>      extern char *optarg;
>      extern int optind;
>      extern int optopt;
>      extern int opterr;
>      extern int optreset;
> 
>      int
>      getopt(int argc, char * const argv[], const char *optstring);
I am afraid we need to keep them as global.